lib/browserstack-service.js

```diff
@@ -32,21 +32,29 @@ class BrowserstackService {
   }
 
   onReload(oldSessionId, newSessionId) {
+
     this.sessionId = newSessionId;
     return this._update(oldSessionId, this._getBody())
-      .then(() => this._printSessionURL());
+      .then(this._printSessionURL())
+      .then(this._reset_failures());
```

Comment:
Since `_reset_failures` is only used once, let's do: `.then(() => { this.failures = 0; })`